### ISSUE TITLE
Using with packaged puppetboard

### ISSUE THREAD

**Issue by eperdeme** [puppet-community/puppet-module-puppetboard#85]
Hello,

I can not see a method of using this module with a packaged version of puppetboard due to the close coupling with Vcsrepo module.

We have no external access to PIP so even if I mirror the puppetboard source the issue still exists.

Ideally I would like to be able to specify package_src = 'git' or for me 'pkg' which changes the way the module handles Vcsrepo and Virtualenv/pip.

**Comment by nibalizer** [comment]
I think this feature is great idea. I have plans to do a major refactor soon, hopefully this feature will make it in.

**Comment by igalic** [comment]
:+1:

**Comment by jbehrends** [comment]
I'd like to see the module also support installing directly from pip.

**Comment by jyaworski** [comment]
@jbehrends do you mean puppetboard as 'packaged' by pip? We currently do a git checkout into a virtualenv.

As for @eperdeme, does setting the git source work for you?

https://github.com/voxpupuli/puppet-module-puppetboard/blob/master/manifests/init.pp#L30-L32

**Comment by jbehrends** [comment]
@jyaworski yes the pip package.  Not everyone's environments allow access to outside resources.  Because of that we mirror the pypi repo locally.

**Comment by jyaworski** [comment]
@jbehrends I'll look into installing from pip directly. This already depends on a python module, so it shouldn't be that difficult.

I feel your pain on not having access to public repos, believe me.

**Comment by mterzo** [comment]
@nibalizer did the refactor happen already?

**Comment by kenyon** [comment]
This was somewhat done in #329.

**Comment by rledousa** [comment]
Hi

Also commented in #202 .  We don't mirror the repo internally, we have an internal pip proxy so our pip.conf looks like:

[global]
index-url = https://pypi.company.com/simple

[search]
index = https://pypi.company.com/

**Comment by gdubicki** [comment]
Hi everyone, especially @rledousa!

Installing from a package is now possible, after https://github.com/voxpupuli/puppet-puppetboard/pull/354 has been merged.

Until the new version of this module is released you get it from GitHub, master/current HEAD hash.

**Comment by rledousa** [comment]
THANK YOU!!!
